### PR TITLE
Fix translation of annotations in opaque-pointers mode

### PR DESCRIPTION
**Pull request by MrSidims**
Signed-off-by: Sidorov, Dmitry <dmitry.sidorov@intel.com>

### CODE REVIEW

File: test/transcoding/opaque-pointer-annotation.ll [hunk 1]
```diff
@@ -0,0 +1,35 @@
+; RUN: llvm-as %s -o %t.bc
```
Comment (MrSidims) [inline test/transcoding/opaque-pointer-annotation.ll:1]: I'll remove this test just updating the one from https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1628 when the latest is merged.

File: test/transcoding/opaque-pointer-annotation.ll [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+; RUN: llvm-spirv %t.bc -spirv-text -o - | FileCheck %s --check-prefix=CHECK-SPIRV
+
+; CHECK-SPIRV: Decorate [[#]] UserSemantic "var_annotation"
```
Comment (MrSidims) [inline test/transcoding/opaque-pointer-annotation.ll:4]: There were empty strings without this patch

**Comment by MrSidims** [comment]
@asudarsa @jcranmer-intel @dwoodwor-intel please take a look

**Review by asudarsa — APPROVED** [review]
LGTM. Thanks

**Comment by jcranmer-intel** [comment]
> @jcranmer-intel please take a look. Are you OK with having two tests opaque and nonopaque?

Sorry for the delay, but this is the normal practice with opaque pointer tests, so there's no issues here.

**Pull request merged**